tests/trainer/test_supporters.py

```diff
@@ -237,3 +241,39 @@ def test_nested_calc_num_data(input_data, compute_func, expected_length):
     calculated_length = _nested_calc_num_data(input_data, compute_func)
 
     assert calculated_length == expected_length
+
+
+@RunIf(min_gpus=2, special=True)
+def test_combined_data_loader_validation_test(tmpdir):
+    """
+    This test makes sure distributed sampler has been properly injected in dataloaders
+    when using CombinedLoader
+    """
+
+    class CustomDataset(Dataset):
+
+        def __init__(self, data):
+            self.data = data
+
+        def __len__(self):
+            return len(self.data)
+
+        def __getitem__(self, index):
+            return self.data[index]
+
+    class CustomDataModule(LightningDataModule):
+
+        def val_dataloader(self) -> CombinedLoader:
+            return CombinedLoader({"a": DataLoader(CustomDataset(range(10)))})
+
+    class CustomModel(BoringModel):
+
+        def validation_step(self, batch, batch_idx):
+            v = batch['a']
+            assert (v + int(self.trainer.global_rank == 1)) % 2 == 0
```

Comment:
I don't understand why we are testing the DistributedSampler in a full special test. 
Would a simpler test do it? We have similar tests in tests/trainer/test_dataloaders

```python
dataloader = CombinedLoader()
trainer = Trainer()
dataloader = trainer.auto_add_sampler(dataloader)
assert all(isinstance(loader.sampler, DistributedSampler) for loader in dataloader.loaders))
```

Reply:
Great comment ! You are definitely right !